### PR TITLE
Modernize GL and add gles support

### PR DESCRIPTION
**Pull request by notspiff**

### CODE REVIEW

**Comment by notspiff** [comment]
@rechi please test gles runtime

**Comment by Rechi** [comment]
Works on android-aarch64.

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,17 +7,33 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR})
 enable_language(CXX)
 
 find_package(Kodi REQUIRED)
-if(NOT WIN32)
+
+if(APP_RENDER_SYSTEM STREQUAL "gl" OR NOT APP_RENDER_SYSTEM)
```
Comment (Rechi) [inline CMakeLists.txt:11]:
```suggestion
if(NOT WIN32 AND (APP_RENDER_SYSTEM STREQUAL "gl" OR NOT APP_RENDER_SYSTEM))
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -7,17 +7,33 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR})
...
   find_package(OpenGL REQUIRED)
   set(DEPLIBS ${OPENGL_LIBRARIES})
+  set(includes ${OPENGL_INCLUDE_DIR})
+  add_definitions(-DHAS_OPENGL)
+elseif(APP_RENDER_SYSTEM STREQUAL "gles")
```
Comment (Rechi) [inline CMakeLists.txt:16]:
```suggestion
elseif(NOT WIN32 AND APP_RENDER_SYSTEM STREQUAL "gles")
```

**Comment by AlwinEsch** [comment]
On my system works as well. Thanks a lot, @Rechi about your comments I create another request.

**Pull request merged**